sdks/python/apache_beam/transforms/combiners.py

```diff
@@ -66,8 +66,21 @@ class Mean(object):
   """Combiners for computing arithmetic means of elements."""
   class Globally(ptransform.PTransform):
     """combiners.Mean.Globally computes the arithmetic mean of the elements."""
+    def __init__(self, has_defaults=True, *args, **kwargs):
+      super(Mean.Globally, self).__init__()
+      self.has_defaults = has_defaults
+      self.args = args
+      self.kwargs = kwargs
+
     def expand(self, pcoll):
-      return pcoll | core.CombineGlobally(MeanCombineFn())
+      if self.has_defaults:
+        return pcoll | core.CombineGlobally(MeanCombineFn())
+      else:
+        return pcoll | core.CombineGlobally(MeanCombineFn()).without_defaults()
+
+    def without_defaults(self):
+      self.has_defaults = False
```

Comment:
Another way could be doing `Mean.Globally(without_defaults=True)`,  but I'd say it's against the standard use of `CombineGlobally().without_defaults()`

Reply:
what robert means in this case is that rather than mutating and returning self like so:
```
def without_defaults(self):
  self.has_defaults = False
  return self
```

It would be more useful to return a new instance with the new property. Somewhat like this:
```
def with_defaults(self, has_defaults=True):
  return Mean.Globally(has_defaults=has_defaults, *self.args, **self.kwargs)
```
It's useful to do this because then instance attributes are immutable, and it's easier to manage instances with immutable attributes.

Reply:
Thanks for the suggestions.

I made it look more like the original `without_defaults()` from [`CombineGlobally`](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/transforms/core.py#L1834)